providers/apis/binance/api_handler.go

```diff
@@ -27,26 +26,32 @@ var _ handlers.APIDataHandler[oracletypes.CurrencyPair, *big.Int] = (*APIHandler
 // for more information about the Binance API, refer to the following link:
 // https://github.com/binance/binance-spot-api-docs/blob/master/rest-api.md#public-api-endpoints
 type APIHandler struct {
-	Config
-	BaseURL string
+	// cfg is the config for the Binance API.
+	cfg config.ProviderConfig
+
+	// invertedMarketCfg is convenience struct that contains the inverted market to currency pair mapping.
+	invertedMarketCfg config.InvertedCurrencyPairMarketConfig
```

Comment:
If almost every Handler is going to include the `cfg` and `invertedCfg` why not just contain them in the same struct?